src/m365/planner/commands/task/task-checklistitem-add.ts

```diff
@@ -0,0 +1,107 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import { accessToken } from '../../../../utils/accessToken';
+import { PlannerTaskDetails } from '@microsoft/microsoft-graph-types';
+import { AxiosRequestConfig } from 'axios';
+import { v4 } from 'uuid';
+import GlobalOptions from '../../../../GlobalOptions';
+import Auth from '../../../../Auth';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+import request from '../../../../request';
+
+interface ExtendedPlannerTaskDetails extends PlannerTaskDetails {
+  '@odata.etag': string;
+}
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  taskId: string;
+  title: string;
+  isChecked?: boolean;
+}
+
+class PlannerTaskChecklistItemAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_CHECKLISTITEM_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new checklist item to a Planner task.';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.isChecked = args.options.isChecked || false;
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (accessToken.isAppOnlyAccessToken(Auth.service.accessTokens[this.resource].accessToken)) {
+      this.handleError('This command does not support application permissions.', logger, cb);
+      return;
+    }
+
+    this
+      .getTaskDetails(args.options.taskId)
+      .then(details => {
+        const etag: string = details['@odata.etag'];
```

Comment:
since, if I understand this properly 😉, we need only the the etag from the task details maybe we could refactor this a bit so that  `getTaskDetails` method would just return the etag as string (maybe then we could have also some better method name).
and with that change we may also remove the `ExtendedPlannerTaskDetails`